src/Workspaces/VisualBasic/Portable/Utilities/DirectiveWalker.vb

```diff
@@ -114,6 +124,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Utilities
             While _regionStack.Count > 0
                 _startEndMap.Add(_regionStack.Pop(), Nothing)
             End While
+
+            While _ifStack.Count > 0
```

Comment:
When we finished the node, we were matching up regions, but we weren't populating the conditional map.